src/fetchers/opensea.js

```diff
@@ -13,58 +13,79 @@ export const fetchCollection = async (chainId, { contract, tokenId }) => {
   try {
     let data;
 
-    try {
-      const url = `${
-        chainId === 1
-          ? process.env.OPENSEA_BASE_URL || "https://api.opensea.io"
-          : "https://testnets-api.opensea.io"
-      }/api/v1/asset/${contract}/${tokenId}`;
+    const url = `${
+      ![4, 5].includes(chainId)
+        ? process.env.OPENSEA_BASE_URL || "https://api.opensea.io"
+        : "https://testnets-api.opensea.io"
+    }/api/v1/events?token_id=${tokenId}&asset_contract_address=${contract}`;
+
+    const assetResponse = await axios.get(url, {
+      headers: ![4, 5].includes(chainId)
+        ? {
+            [process.env.OPENSEA_API_HEADER ?? "X-API-KEY"]: process.env.OPENSEA_API_KEY.trim(),
+            Accept: "application/json",
+          }
+        : {
+            Accept: "application/json",
+          },
+    });
+
+    data = assetResponse.data.asset_events && assetResponse.data.asset_events[0].asset;
```

Comment:
small comment, you might be able to do :
`data = assetResponse.data.asset_events[0]?.asset;`